test/mitmproxy/test_proxy.py

```diff
@@ -80,6 +82,37 @@ def test_simple(self):
 
 class TestConnectionHandler:
 
+    def check_proxy_for_mode(self, mode):
+        """
+        Checks that we get correct log message in specific mode
+        """
+        log_msg = "The proxy shall not connect to itself."
+        opts = options.Options(mode=mode)
+        pconf = config.ProxyConfig(opts)
+
+        channel = mock.Mock()
+
+        def tell(mtype, m):
+            if m.level == "warn":
```

Comment:
What happens in this test if there is no "warn" message?
Maybe set a bool flag `warning_triggered=True` and then at the end of the test assert that.

Reply:
@Kriechi yes, you're right!